src/app/zap-templates/zcl/data-model/chip/icd-management-cluster.xml

```diff
@@ -16,55 +16,67 @@ limitations under the License.
-            <description>ExpectedClients</description>
+        <attribute side="server" code="0x03" define="REGISTERED_CLIENTS" type="ARRAY" entryType="MonitoringRegistrationStruct" writable="false" optional="true" isNullable="false">
+            <description>RegisteredClients</description>
             <access op="read" privilege="administer"/>
         </attribute>
-
-        <attribute side="server" code="0x04" define="ICD_COUNTER" type="INT32U" writable="false" optional="false" isNullable="false">
+        <attribute side="server" code="0x04" define="ICD_COUNTER" type="INT32U" writable="false" optional="true" isNullable="false">
             <description>ICDCounter</description>
             <access op="read" privilege="administer"/>
         </attribute>
+        <attribute side="server" code="0x05" define="CLIENTS_SUPPORTED_PER_FABRIC" type="INT16U" min="1" default="1" writable="false" optional="true" isNullable="false">ClientsSupportedPerFabric</attribute>
 
-
-        <command source="client" code="0x00" name="RegisterClientMonitoring" response="RegisterClientMonitoringResponse" isFabricScoped="true" optional="false">
+        <command source="client" code="0x00" name="RegisterClient" response="RegisterClientResponse" isFabricScoped="true" optional="true">
             <description> Register a client to the end device </description>
-            <arg name="ClientNodeID" type="node_id"/>
-            <arg name="Key" type="OCTET_STRING" length="16"/>
+            <arg name="CheckInNodeID" type="node_id" optional="false"/>
+            <arg name="MonitoredSubject" type="INT64U" optional="false"/>
+            <arg name="Key" type="OCTET_STRING" length="16" optional="false"/>
+            <arg name="VerificationKey" type="OCTET_STRING" length="16" optional="true"/>
             <access op="invoke" privilege="manage"/>
         </command>
 
-        <command source="server" code="0x01" name="RegisterClientMonitoringResponse" optional="false" disableDefaultResponse="true">
-            <description> RegisterClientMonitoring response command </description>
-            <arg name="Status" type="ENUM8"/>
-            <arg name="ICDCounter" type="INT32U" isNullable="true"/>
+        <command source="server" code="0x01" name="RegisterClientResponse" optional="false" disableDefaultResponse="false">
+            <description> RegisterClient response command </description>
+            <arg name="ICDCounter" type="INT32U" isNullable="false"/>
         </command>
 
-        <command source="client" code="0x02" name="UnregisterClientMonitoring" isFabricScoped="true" optional="false">
+        <command source="client" code="0x02" name="UnregisterClient" isFabricScoped="true" optional="true">
             <description> Unregister a client from an end device </description>
-            <arg name="ClientNodeID" type="node_id"/>
+            <arg name="CheckInNodeID" type="node_id"/>
```

Comment:
Spec says "subject-id" here, but that looks like a bug in the spec....

Reply:
You are correct. The change in the spec is tracked by https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/7008